great_expectations/data_context/store/gx_cloud_store_backend.py

```diff
@@ -327,8 +327,13 @@ def _put(self, id: str, value: Any) -> bool:
                 response_status_code = response.status_code
 
             response.raise_for_status()
-            return True
-
+            try:
+                response_json = response.json()
+            except JSONDecodeError:
+                return True
```

Comment:
yeah. this method is used by all store implementations, and other `update` calls (like Checkpoint) do return a boolean. fixing the store implementation is explicitly out of scope, but let's improve this soon.

Reply:
Can you add a "why" comment on this!

Reply:
even better - updated this to use the returned HTTP code instead of exceptions. It does require that the backend use the correct code, but is definitely easier to reason about